components/github/actions/create-pull-request/create-pull-request.mjs

```diff
@@ -0,0 +1,139 @@
+      `),
+      type: "string",
+    },
+    base: {
+      label: "Base Branch",
+      description: toSingleLineString(`
+      The name of the branch you want the changes pulled into.
+      This should be an existing branch on the current repository.
+      You cannot submit a pull request to one repository that requests a merge to a base of another repository.
+      `),
+      type: "string",
+    },
+    headRepo: {
+      label: "Head Repository's Name",
+      description: toSingleLineString(`
+      The name of the repository where the changes in the pull request were made.
+      This field is required for cross-repository pull requests if both repositories are owned by the same organization.
+      `),
+      type: "string",
+      optional: true,
+    },
+    body: {
+      label: "Body",
+      description: "The contents of the pull request.",
+      type: "string",
+      optional: true,
+    },
+    maintainerCanModify: {
+      label: "Maintainer Can Modify",
+      description: "Indicates whether [maintainers can modify](https://docs.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) the pull request.",
+      type: "boolean",
+      optional: true,
+    },
+    draft: {
+      label: "Is Draft",
+      description: toSingleLineString(`
+      Indicates whether the pull request is a draft.
+      See "[Draft Pull Requests](https://docs.github.com/articles/about-pull-requests#draft-pull-requests)" in the GitHub Help documentation to learn more.
+      `),
+      type: "boolean",
+      optional: true,
+    },
+    convertCurrentIssue: {
+      type: "string",
+      label: "Do you want to convert an issue to a pull request, or create a new pull request?",
+      description: toSingleLineString(`
+      If you convert an issue to a pull request the issue's title, body, and comments will become the title, body, and comments on the new pull request.
+      Otherwise you can manually enter a title, and body for a new pull request.
+      `),
+      options: [
+        "Convert issue to pull request",
+        "Create a new pull request",
+      ],
+      reloadProps: true,
+    },
+  },
+  async additionalProps() {
+    const props = {};
+    if (this.convertCurrentIssue === "Convert issue to pull request") {
+      props.issue = {
+        label: "Issue",
+        description: toSingleLineString(`
+        An issue in the repository to convert to a pull request.
+        The issue title, body, and comments will become the title, body, and comments on the new pull request.
+        `),
+        type: "integer",
+        min: 1,
+      };
+    } else if (this.convertCurrentIssue === "Create a new pull request") {
+      props.title = {
+        label: "Title",
+        description: "The title of the new pull request.",
+        type: "string",
+      };
+    }
+
+    return props;
+  },

```

Comment:
The prop `issue` should have a dropdown list of options, but since async options can be tricky to implement within additionalProps, I would suggest listing both `issue` and `title` as optional props. Then if neither are entered, you can throw a ConfigurationError with the relevant message.
```suggestion
    title: {
      label: "Title",
      description: "The title of the new pull request.",
      type: "string",
      optional: true,
    },
    issue: {
      propDefinition: [
        github,
        "issueNumber",
        (c) => ({
          repoFullname: c.repoFullname,
        }),
      ],
      label: "Issue",
      description: toSingleLineString(`
        An issue in the repository to convert to a pull request.
        The issue title, body, and comments will become the title, body, and comments on the new pull request.
        `),
      min: 1,
      optional: true,
    },
  },
```